File: src/openstaad_mcp/staad_skills/staad-properties/scripts/assign-plate-thickness.py
# assign-plate-thickness.py
# Assigns a uniform thickness to all plates in the model.
# Always retrieve actual plate IDs — never assume they start at 1.
# CreatePlateThicknessProperty requires a list of 4 floats (one per corner node).

geo = staad.Geometry
prop = staad.Property

# Get actual plate IDs
plate_ids = list(geo.GetPlateList())
print(f'Plates found: {plate_ids}')

# Create thickness property — 4 floats required, one per corner node
t = 3.937   # 100 mm expressed in inches (English model); use meters for Metric
thick_id = prop.CreatePlateThicknessProperty([t, t, t, t])
print(f'Created thickness property ID: {thick_id} (t={t} in)')

# Assign to all plates
prop.AssignPlateThickness(plate_ids, thick_id)
print(f'Assigned thickness {t} in to {len(plate_ids)} plates')

File: src/openstaad_mcp/staad_skills/staad-properties/scripts/assign-plate-thickness.js
// assign-plate-thickness.js
// Assigns a uniform thickness to all plates in the model.
// Always retrieve actual plate IDs — never assume they start at 1.
// CreatePlateThicknessProperty requires a list of 4 floats (one per corner node).

const geo = staad.Geometry;
const prop = staad.Property;

// Get actual plate IDs
const plateIds = geo.GetPlateList();
console.log(`Plates found: ${JSON.stringify(plateIds)}`);

// Create thickness property — 4 floats required, one per corner node
const t = 3.937;   // 100 mm expressed in inches (English model); use meters for Metric
const thickId = prop.CreatePlateThicknessProperty([t, t, t, t]);
console.log(`Created thickness property ID: ${thickId} (t=${t} in)`);

// Assign to all plates
prop.AssignPlateThickness(plateIds, thickId);
console.log(`Assigned thickness ${t} in to ${plateIds.length} plates`);
